corehq/form_processor/models.py

```diff
@@ -413,14 +413,31 @@ def write_content(self, content):
         if not self.name:
             raise InvalidAttachment("cannot save attachment without name")
 
+        content_readable = content
+        if isinstance(content, basestring):
+            content_readable = StringIO(content)
```

Comment:
this should actually always be a string so could potentially just more the `if`. I just added it as a safeguard